cmd/serverless-init/log/log.go

```diff
@@ -92,11 +96,48 @@ func SetupLog(conf *Config, tags map[string]string) {
 }
 
 func (cw *CustomWriter) Write(p []byte) (n int, err error) {
-	fmt.Print(string(p))
-	Write(cw.LogConfig, p, cw.IsError)
+	if len(p) > maxBufferSize {
+		p = p[:maxBufferSize]
+	}
+
+	if !cw.IsDotnet {
+		fmt.Print(string(p))
+		Write(cw.LogConfig, p, cw.IsError)
+		return len(p), nil
+	}
+
+	// .NET specific behavior
+
+	// Prevent buffer overflow, flush the buffer if writing the current chunk
+	// will exceed maxBufferSize
+	if cw.LineBuffer.Len()+len(p) > maxBufferSize {
+		fmt.Print(string(p))
```

Comment:
in the case of an overflow, does it make more sense to `Print` or `Println` the log? `Print` preserves what's expected, though `Println` would reflect what shows up in Datadog

Reply:
I think Print makes sense. Although, this makes me think, should we always just do fmt.Print immediately for each payload, (so we are completely passing through the underlying behaviour), and only apply the buffering for the Write call?

Reply:
On second thought the `println` later doesn't really make sense, it should just be a passthrough. I'll make that change